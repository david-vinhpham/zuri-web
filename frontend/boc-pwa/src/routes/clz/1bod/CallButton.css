/**
 * Home Credit (https://www.homecredit.vn/)
 *
 * Copyright Â© 2018-present HomeCredit, LLC. All rights reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE.txt file in the root directory of this source tree.
 */

@import '../../../components/variables.css';

:root {
  --max-modal-height: 380px;
  --max-modal-img-w: 300px;
}

.welcomeXs {
  display: none;
}

.welcome {
  z-index: calc(var(--z-index-high) - 1);
}

.phone {
  border: var(--b-space-3xs) solid var(--c-brand);
  height: calc(var(--unit) * 3);
  width: calc(var(--unit) * 3);
  border-radius: 50%;
  margin: 0 var(--b-space-2xs) var(--b-space-2xs) 0;
  background: linear-gradient(to bottom, var(--c-brand-light), var(--c-brand-dark));
  display: flex;
  justify-content: center;
  align-items: center;
  cursor: pointer;
  box-shadow: -9px 8px 15px 0 rgba(240, 14, 47, 0.2), -4px 5px 17px 0 rgba(0, 0, 0, 0.2);

  & span {
    margin: 0;
  }

  & img {
    width: calc(var(--unit) * 1.4);
  }
}

.backdrop {
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  position: fixed;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  background: rgba(0, 0, 0, 0.4);
  z-index: var(--z-index-highest);
}

.modalContent {
  max-width: var(--max-modal-height);
  max-height: var(--max-modal-height);
  text-align: center;
  background: var(--c-ui-lighter);
}

.hotNumber {
  font-size: var(--font-size-big);
  font-weight: var(--font-weight-bold);
  color: var(--c-brand);
}

.closeModalButton {
  border-radius: 0;
}

@media (--xs) {
  .welcomeXs {
    display: flex;
    justify-content: flex-end;
  }

  .welcome {
    display: none;
  }
}
