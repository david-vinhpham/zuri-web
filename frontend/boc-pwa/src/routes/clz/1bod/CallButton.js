import React, { Component } from 'react';
import PropTypes from 'prop-types';
import withStyles from 'isomorphic-style-loader/lib/withStyles';
import { Button } from 'homecredit-ui';
import classnames from 'classnames';
import { sendGAEvent } from 'utils/ga';

import s from './CallButton.css';
import iconCall from './images/call.svg';

class CallButton extends Component {
  state = {
    isShowingHotLine: false,
  };

  handleShowModal = () => {
    this.setState({ isShowingHotLine: true });
    this.trackingCallButton();
  };

  handleCloseModal = () => {
    this.setState({ isShowingHotLine: false });
  };

  trackingCallButton = () => {
    sendGAEvent('Click_call_hotline', 'Hotline', 'Hotline');
  };

  render() {
    const { className } = this.props;
    const { isShowingHotLine } = this.state;

    return (
      <>
        <a
          href="tel:*6868"
          className={classnames(s.welcomeXs, className)}
          onClick={this.trackingCallButton}
        >
          <div className={s.phone}>
            <img src={iconCall} alt="call button" />
          </div>
        </a>
        <div
          className={classnames(s.welcome, className)}
          onClick={this.handleShowModal}
        >
          <div className={s.phone}>
            <img src={iconCall} alt="call button" />
          </div>
        </div>

        {isShowingHotLine && (
          <div className={s.backdrop}>
            <div className={s.modalContent}>
              <p>
                Hãy gọi <span className={s.hotNumber}>*6868</span> hoặc{' '}
                <span className={s.hotNumber}>(028) 38 999 666</span> để được
                hướng dẫn thêm!
              </p>
              <Button
                stretch
                className={s.closeModalButton}
                onClick={this.handleCloseModal}
              >
                Đóng
              </Button>
            </div>
          </div>
        )}
      </>
    );
  }
}

CallButton.propTypes = {
  className: PropTypes.string,
};

CallButton.defaultProps = {
  className: '',
};

export default withStyles(s)(CallButton);
