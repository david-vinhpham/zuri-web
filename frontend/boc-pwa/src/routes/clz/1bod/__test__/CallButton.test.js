/* eslint-env jest */
import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import CallButton from '../CallButton';

describe('Call Button', () => {
  it('should render call button with default props', () => {
    const wrapper = renderer.create(<CallButton />);
    const snapshot = wrapper.toJSON();

    expect(snapshot).toMatchSnapshot();
  });

  it('should show hotline modal when click button', () => {
    const wrapper = shallow(<CallButton />);

    wrapper.find('.welcome').simulate('click');

    expect(wrapper.state().isShowingHotLine).toBe(true);
  });
});
