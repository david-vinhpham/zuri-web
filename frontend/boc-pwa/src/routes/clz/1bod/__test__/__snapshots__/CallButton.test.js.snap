// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Call Button should render call button with default props 1`] = `
Array [
  <a
    className="welcomeXs"
    href="tel:*6868"
    onClick={[Function]}
  >
    <div
      className="phone"
    >
      <img
        alt="call button"
        src="call.svg"
      />
    </div>
  </a>,
  <div
    className="welcome"
    onClick={[Function]}
  >
    <div
      className="phone"
    >
      <img
        alt="call button"
        src="call.svg"
      />
    </div>
  </div>,
]
`;
